crm_rma_stock_location/product.py

```diff
@@ -0,0 +1,153 @@
+                ctx['location'] = rma_id
+        else:
+            location_ids = set()
+            wids = warehouse_obj.search([])
+            if not wids:
+                return
+            for wh in wids:
+                if wh.lot_rma_id:
+                    location_ids.add(wh.lot_rma_id.id)
+            if not location_ids:
+                return
+            ctx['location'] = list(location_ids)
+
+        domain_products = [('product_id', 'in', [self.id])]
+        domain_quant, domain_move_in, domain_move_out = \
+            self.with_context(ctx)._get_domain_locations()
+        domain_move_in += self.with_context(ctx)._get_domain_dates() + \
+            [('state',
+              'not in',
+              ('done', 'cancel', 'draft'))] + domain_products
+        domain_move_out += self.with_context(ctx)._get_domain_dates() + \
+            [('state',
+              'not in',
+              ('done', 'cancel', 'draft'))] + domain_products
+        domain_quant += domain_products
+        if context.get('lot_rma_id'):
+            if context.get('lot_rma_id'):
+                domain_quant.append(('lot_rma_id', '=', context['lot_rma_id']))
+            moves_in = []
+            moves_out = []
+        else:
+            moves_in = self.env['stock.move'].\
+                with_context(ctx).read_group(domain_move_in,
+                                             ['product_id',
+                                              'product_qty'],
+                                             ['product_id'])
+            moves_out = self.env['stock.move'].\
+                with_context(ctx).read_group(domain_move_out,
+                                             ['product_id',
+                                              'product_qty'],
+                                             ['product_id'])
+
+        quants = self.env['stock.quant'].\
+            with_context(ctx).read_group(domain_quant,
+                                         ['product_id',
+                                          'qty'],
+                                         ['product_id'])
+
+        quants = dict([(item.get('product_id')[0],
+                        item.get('qty')) for item in quants])
+
+        moves_in = dict([(item.get('product_id')[0],
+                          item.get('product_qty')) for item in moves_in])
+
+        moves_out = dict([(item.get('product_id')[0],
+                           item.get('product_qty')) for item in moves_out])
+
+        self.rma_qty_available = \
+            float_round(quants.get(self.id, 0.0),
+                        precision_rounding=self.uom_id.rounding)
+
+        self.rma_virtual_available = \
+            float_round(quants.get(self.id, 0.0) +
+                        moves_in.get(self.id, 0.0) -
+                        moves_out.get(self.id, 0.0),
+                        precision_rounding=self.uom_id.rounding)
+
+    rma_qty_available = fields.Float(compute='_rma_product_available',
```

Comment:
Maybe without storing, the performance slows down, but you have to put correctly depends (maybe creating auxiliary fields).